main.tf

```diff
@@ -94,19 +94,47 @@ resource "google_dns_managed_zone" "public" {
   project     = var.project_id
   name        = var.name
   dns_name    = var.domain
-  description = "Terraform-managed zone."
+  description = var.description
   visibility  = "public"
+
+  dynamic "dnssec_config" {
+    for_each = var.dnssec_config == {} ? [] : list(var.dnssec_config)
+    content {
```

Comment:
This has taken me a few minutes to understand. :) Would it be possibly be more legible if you used something like

```
iterator = config
content {
  kind = lookup(config, "kind", "dns#managedZoneDnsSecConfig")
```

And maybe instead of using `lookup` you could `merge` a map containing defaults with the map from variables in the `for_each`.

Reply:
I did do the iterator one.
For the "merge", I'm not really sure it will improve readability